hasher-matcher-actioner/hmalib/lambdas/api/matches.py

```diff
@@ -306,25 +309,15 @@ def for_hash(request: MatchesForHashRequest) -> MatchesForHashResponse:
         NOTE: currently metadata returned will not be written to the dynamodb table
         unlike in the case of a pipeline match based on submissions.
         """
-        signal_type = None
-        if request.signal_type == "pdq":
-            # todo translate in MatchesForHashRequest and extend to cover MD5
-            signal_type = PdqSignal
-
-        if not signal_type:
-            # only support PDQ at the moment
-            bottle.response.status = 400
-            return MatchesForHashResponse([], request.signal_value)
 
         matches = _get_matcher(indexes_bucket_name).match(
-            signal_type, request.signal_value
+            request.signal_type, request.signal_value
         )
 
         match_objects = []
```

Comment:
nit: type this? It is not immediately apparent what the type for this will be. Esp with the `.extend()` thing we're doing a few lines down.